src/utilities/lexicographicSortSchema.js

```diff
@@ -74,6 +75,20 @@ export function lexicographicSortSchema(schema: GraphQLSchema): GraphQLSchema {
     return maybeType && replaceNamedType(maybeType);
   }
 
+  function replaceResolvedType(
+    maybeTypeResolver: ?GraphQLTypeResolver<mixed, mixed>,
+  ) {
+    return (
+      maybeTypeResolver &&
+      ((...args) =>
+        Promise.resolve(maybeTypeResolver(...args)).then((resolvedType) =>
```

Comment:
An important property of `execute` is that it returns only if at least one resolver return promise.
This change will convert sync schema into async.
Please use `isPromise`